docs/core/seeding.rst

```diff
@@ -93,77 +101,103 @@ only once, the circle calculation is spread out over the three loops.
-
-     // contains parameters required to calculate circle with linear equation
-     // ...for bottom-middle
-     std::vector<LinCircle> linCircleBottom;
-     // ...for middle-top
-     std::vector<LinCircle> linCircleTop;
-     transformCoordinates(compatBottomSP, *spM, true, linCircleBottom);
-     transformCoordinates(compatTopSP, *spM, false, linCircleTop);
-
-     // create vectors here to avoid reallocation in each loop
-     std::vector<const InternalSpacePoint<external_spacepoint_t>*> topSpVec;
-     std::vector<float> curvatures;
-     std::vector<float> impactParameters;
-
-     std::vector<std::pair<
-         float, std::unique_ptr<const InternalSeed<external_spacepoint_t>>>>
-         seedsPerSpM;
-     size_t numBotSP = compatBottomSP.size();
-     size_t numTopSP = compatTopSP.size();
-
-     for (size_t b = 0; b < numBotSP; b++) {
-       auto lb = linCircleBottom[b];
-       float Zob = lb.Zo;
-       float cotThetaB = lb.cotTheta;
-       float Vb = lb.V;
-       float Ub = lb.U;
-       float ErB = lb.Er;
-       float iDeltaRB = lb.iDeltaR;
-
-       // 1+(cot^2(theta)) = 1/sin^2(theta)
-       float iSinTheta2 = (1. + cotThetaB * cotThetaB);
-       // calculate max scattering for min momentum at the seed's theta angle
-       // scaling scatteringAngle^2 by sin^2(theta) to convert pT^2 to p^2
-       // accurate would be taking 1/atan(thetaBottom)-1/atan(thetaTop) <
-       // scattering
-       // but to avoid trig functions we approximate cot by scaling by
-       // 1/sin^4(theta)
-       // resolving with pT to p scaling --> only divide by sin^2(theta)
-       // max approximation error for allowed scattering angles of 0.04 rad at
-       // eta=infinity: ~8.5%
-       float scatteringInRegion2 = m_config.maxScatteringAngle2 * iSinTheta2;
-       // multiply the squared sigma onto the squared scattering
-       scatteringInRegion2 *=
-           m_config.sigmaScattering * m_config.sigmaScattering;
-
-       // clear all vectors used in each inner for loop
-       topSpVec.clear();
-       curvatures.clear();
-       impactParameters.clear();
-       for (size_t t = 0; t < numTopSP; t++) {
-         auto lt = linCircleTop[t];
-
-         // protects against division by 0
-         float dU = lt.U - Ub;
-         if (dU == 0.) {
-           continue;
-         }
-         // A and B are evaluated as a function of the circumference parameters
-         // x_0 and y_0
-         float A = (lt.V - Vb) / dU;
-         float S2 = 1. + A * A;
-         float B = Vb - A * Ub;
-         float B2 = B * B;
-         // sqrt(S2)/B = 2 * helixradius
-         // calculated radius must not be smaller than minimum radius
-         if (S2 < B2 * m_config.minHelixDiameter2) {
-           continue;
-         }
-         // 1/helixradius: (B/sqrt(S2))/2 (we leave everything squared)
-         float iHelixDiameter2 = B2 / S2;
+	for (auto spM : middleSPs) {
+   
+		... // compatibility cuts between SP duplets
+
```

Comment:
The problems with these code listings is that they have the potential to become outdated very quickly.
Do you see any way you could break this up somehow and describe the sequence in words, while linking to the various functions that are called using the sphinx syntax?